pandas/core/ops/__init__.py

```diff
@@ -338,27 +337,28 @@ def column_op(a, b):
             # Note: we do not do this unconditionally as it may be lossy or
             #  expensive for EA dtypes.
             right = np.asarray(right)
-
-            def column_op(a, b):
-                return {i: func(a.iloc[:, i], b[i]) for i in range(len(a.columns))}
-
         else:
+            right = right._values
 
-            def column_op(a, b):
-                return {i: func(a.iloc[:, i], b.iloc[i]) for i in range(len(a.columns))}
```

Comment:
doesnt need to be in this PR, but since you're attention is already focused on this line: this can be incorrect if `b` is e.g. Categorical, since `b.iloc[i]` will have the scalar behavior instead of the Categorical (usually raising) behavior.  (This is also a place where NaT causes issues, and the idea of only-one-NA causes me nightmares)

The main idea I've had here is to use `b.iloc[[i]]` instead of `b.iloc[i]`, haven't gotten around to implementing it.  Does this change make it easier to address the issue?

Reply:
when using something like `b.iloc[[i]]`, you would need to handle broadcasting here as well (since the array ops expect either same length array or scalar). So since that is a pre-existing issue, I would leave that for a dedicated PR. 
Is there an issue for this?

Reply:
I dont think there's an issue for this.  If I had found one, i would have added the Numeric label